configuration-service/common/projects_materialized_view.go

```diff
@@ -206,3 +210,60 @@ func (mv *projectsMaterializedView) DeleteService(project string, stage string,
 	fmt.Println("Deleted service " + service + " from stage " + stage + " in project " + project)
 	return nil
 }
+
+func (mv *projectsMaterializedView) UpdateEventOfService(keptnBase *keptn.KeptnBase, eventType string, keptnContext string, eventID string) error {
+	existingProject, err := mv.GetProject(keptnBase.Project)
+	if err != nil {
+		fmt.Println("Could not update service " + keptnBase.Service + " in stage " + keptnBase.Stage + " in project " + keptnBase.Project + ". Could not load project: " + err.Error())
+		return err
+	}
+
+	contextInfo := &models.EventContext{
+		EventID:      eventID,
+		KeptnContext: keptnContext,
+		Time:         strconv.FormatInt(time.Now().UnixNano(), 10),
+	}
+	err = updateServiceInStage(existingProject, keptnBase.Stage, keptnBase.Service, func(service *models.ExpandedService) error {
+		if service.LastEventTypes == nil {
+			service.LastEventTypes = map[string]models.EventContext{}
+		}
+		if eventType == keptn.DeploymentFinishedEventType {
+			if keptnBase.Image != nil && keptnBase.Tag != nil {
+				service.DeployedImage = *keptnBase.Image + ":" + *keptnBase.Tag
+			}
+		}
+		service.LastEventTypes[eventType] = *contextInfo
+		return nil
+	})
+
+	if err != nil {
+		fmt.Println("Could not update image of service " + keptnBase.Service + ": " + err.Error())
+		return err
+	}
+	err = mv.updateProject(existingProject)
+	if err != nil {
+		fmt.Println("Could not update " + keptnBase.Project + ": " + err.Error())
+		return err
+	}
+	return nil
+}
+
+type serviceUpdateFunc func(service *models.ExpandedService) error
+
+func updateServiceInStage(project *models.ExpandedProject, stage string, service string, fn serviceUpdateFunc) error {
+	for _, stg := range project.Stages {
+		if stg.StageName == stage {
+			serviceIndex := -1
+			for idx, svc := range stg.Services {
+				if svc.ServiceName == service {
+					serviceIndex = idx
+				}
+			}
+			if serviceIndex < 0 {
+				return errors.New("service not found")
+			}
+			return fn(stg.Services[serviceIndex])
+		}
```

Comment:
Is the case possible that no stage is found? 
I.e. should we add `return errors.New("stage not found")`?

Reply:
that is handled by the return statement at the end of the function. i have updated the error message for that case to indicate that the stage has not been found